core/src/main/scala/akka/kafka/javadsl/Producer.scala

```diff
@@ -152,6 +153,116 @@ object Producer {
+   *
+   * - [[akka.kafka.ProducerMessage.PassThroughMessage PassThroughMessage]] does not publish anything, but commits the offset
+   *
+   * Note that there is a risk that something fails after publishing but before
+   * committing, so it is "at-least once delivery" semantics.
+   */
+  def committableSink[K, V, IN <: Envelope[K, V, ConsumerMessage.Committable]](
+      producerSettings: ProducerSettings[K, V],
+      committerSettings: CommitterSettings
+  ): Sink[IN, CompletionStage[Done]] =
+    scaladsl.Producer
+      .committableSink(producerSettings, committerSettings)
+      .mapMaterializedValue(_.toJava)
+      .asJava
+
+  /**
+   * Create a sink that is aware of the [[ConsumerMessage.CommittableOffset committable offset]]
+   * from a [[Consumer.committableSource]]. The offsets are batched and committed regularly.
+   *
+   * It publishes records to Kafka topics conditionally:
+   *
+   * - [[akka.kafka.ProducerMessage.Message Message]] publishes a single message to its topic, and commits the offset
+   *
+   * - [[akka.kafka.ProducerMessage.MultiMessage MultiMessage]] publishes all messages in its `records` field, and commits the offset
+   *
+   * - [[akka.kafka.ProducerMessage.PassThroughMessage PassThroughMessage]] does not publish anything, but commits the offset
+   *
+   * Note that there is a risk that something fails after publishing but before
+   * committing, so it is "at-least once delivery" semantics.
+   *
+   * Uses a shared a Kafka Producer instance.
+   */
+  def committableSink[K, V, IN <: Envelope[K, V, ConsumerMessage.Committable]](
+      producerSettings: ProducerSettings[K, V],
+      committerSettings: CommitterSettings,
+      producer: org.apache.kafka.clients.producer.Producer[K, V]
+  ): Sink[IN, CompletionStage[Done]] =
+    scaladsl.Producer
+      .committableSink(producerSettings, committerSettings, producer)
+      .mapMaterializedValue(_.toJava)
+      .asJava
+
+  /**
+   * Create a sink that is aware of the [[ConsumerMessage.Committable committable offset]] passed as
+   * context from a [[Consumer.sourceWithOffsetContext]]. The offsets are batched and committed regularly.
+   *
+   * It publishes records to Kafka topics conditionally:
+   *
+   * - [[akka.kafka.ProducerMessage.Message Message]] publishes a single message to its topic, and commits the offset
+   *
+   * - [[akka.kafka.ProducerMessage.MultiMessage MultiMessage]] publishes all messages in its `records` field, and commits the offset
+   *
+   * - [[akka.kafka.ProducerMessage.PassThroughMessage PassThroughMessage]] does not publish anything, but commits the offset
+   *
+   * Note that there is a risk that something fails after publishing but before
+   * committing, so it is "at-least once delivery" semantics.
+   */
+  def sinkWithOffsetContext[K, V, IN <: Envelope[K, V, _], C <: CommittableOffset](
```

Comment:
Should it be called `committingSinkWithOffsetContext`?  I know it's a mouthful..

Reply:
I changed to `sinkWithCommitting` and `sinkWithCommittingOffsetContext` instead.